tools/@aws-cdk/spec2cdk/lib/cdk/resource-decider.ts

```diff
@@ -58,39 +61,73 @@ export class ResourceDecider {
     // has both `Arn` and `ResourceArn`, and we want to select the `Arn` property.
     const possibleArnNames = ['Arn', 'ResourceArn', `${this.resource.name}Arn`];
     for (const arn of possibleArnNames) {
-      const att = this.classAttributeProperties.filter((a) => a.propertySpec.name === attributePropertyName(arn));
-      const prop = this.propsProperties.filter((p) => p.propertySpec.name === propertyNameFromCloudFormation(arn));
-      if (att.length > 0 || prop.length > 0) {
-        return att[0] ? att[0].propertySpec : prop[0].propertySpec;
-      }
+      const att = this.classAttributeProperties.find((a) => a.propertySpec.name === attributePropertyName(arn));
+      if (att) { return att.propertySpec; }
+      // TODO: double check -- if an arn is a property, not an attribute, and not a primaryIdentifier
+      // it is not getting returned by cloudformation in any way, right?
+      // const prop = this.propsProperties.find((p) => p.propertySpec.name === propertyNameFromCloudFormation(arn));
+      // if (prop) {
+      //   return this.convertPropertySpecToRefAttribute(prop.propertySpec);
+      // }
     }
     return;
   }
 
+  private convertPropertySpecToRefAttribute(propSpec: PropertySpec): PropertySpec {
+    return {
+      ...propSpec,
+      name: attributePropertyName(propSpec.name[0].toUpperCase() + propSpec.name.slice(1)),
+      docs: {
+        ...propSpec.docs,
+        remarks: propSpec.docs?.remarks?.concat(['\n', '@cloudformationAttribute Ref'].join('\n')),
+      },
+      immutable: true,
+    };
+  }
+
   private convertPrimaryIdentifier() {
-    for (const cfnName of this.resource.primaryIdentifier ?? []) {
+    if (this.resource.primaryIdentifier === undefined) { return; }
+    for (let i = 0; i < (this.resource.primaryIdentifier).length; i++) {
+      const cfnName = this.resource.primaryIdentifier[i];
       const att = this.findAttributeByName(attributePropertyName(cfnName));
       const prop = this.findPropertyByName(propertyNameFromCloudFormation(cfnName));
       if (att) {
         this.primaryIdentifier.push(att);
```

Comment:
What is `this.primaryIdentifier` used for?

Aren't we just saying that for primary identifier parts that are ALSO attributes, we don't need to do anything because they will already be initialized as part of some other code somewhere?